code/datums/ai/ai.dm

```diff
@@ -1,5 +1,17 @@
 /datum/ai
 	var/name
+	var/mob/living/body		// The parent mob we control.
+	var/waiting	= 	0		// The number of process ticks we've been waiting.
+	var/wait_for = 	1		// How many process ticks to wait before processing.
+	
+/datum/ai/New(var/mob/living/_body)
+	body = _body
 
-/datum/ai/proc/process()
-	return PROCESS_KILL
+/datum/ai/proc/can_process()
```

Comment:
A can-checking proc is typically pure

Reply:
I don't understand this comment.

Reply:
A method that checks if something can/should be done typically shouldn't alter state
Instead consider the more usual:
```
/datum/ai
    var/next_process = 0
    var/wait_for = 1

/datum/ai/proc/can_process()
    return world.time => next_process

/datum/ai/proc/process()
    if(can_process()
        next_process = world.time + wait_for
        do_process()
```

Reply:
Valid feedback.